source/Halibut/Diagnostics/LogCreators/ICreateNewILog.cs

```diff
@@ -0,0 +1,9 @@
+using System;
+
+namespace Halibut.Diagnostics.LogCreators
+{
+    public interface ICreateNewILog
```

Comment:
Do we really need this new Creator concept? It seems like were making this area very complicated

Reply:
Is just rebranding a builder as a different name?

Reply:
I didn't initially have the CreateNewLog concept, but @sburmanoctopus convinced me we really have two things a `ILogFactory` which returns logs which could be retained (cached) and another thing that always makes new things. Without this concept, the result was types that wanted to express the idea that they need to be cached/retained.

Going down this path of separating things that always make new ones from things that get you a ILog seemed to force users down the right path.